lit_gpt/config.py

```diff
@@ -40,15 +43,29 @@ class Config:
     # credit https://arxiv.org/pdf/2305.13245.pdf
     n_query_groups: Optional[int] = None
     shared_attention_norm: bool = False
+    _norm_class: str = "LayerNorm"
+    norm_eps: float = 1e-5
+    _mlp_class: str = "GptNeoxMLP"
+    init_std: float = 0.02
+    intermediate_size: Optional[int] = None
 
     def __post_init__(self):
+        # error checking
+        assert self.n_embd % self.n_head == 0
+        # vocab size should be a power of 2 to be optimal on hardware. compute the closest value
         if self.padded_vocab_size is None:
             self.padded_vocab_size = find_multiple(self.vocab_size, self.padding_multiple)
-        assert self.n_embd % self.n_head == 0
+        # compute the number of query groups
         if self.n_query_groups is not None:
             assert self.n_head % self.n_query_groups == 0
         else:
             self.n_query_groups = self.n_head
+        # compute the intermediate size for MLP if not set
+        if self.intermediate_size is None:
+            if self._mlp_class == "LLaMAMLP":
+                raise ValueError("The config needs to set the `intermediate_size`")
```

Comment:
I couldn't find the intermediate size formula used by open llama, so asking to pass it explicitly. It's defined in the hugginface configs